### DISCUSSION TITLE
Is there problems with train signaling?

### DISCUSSION THREAD

**Discussion by hkoskenv** [voxeltycoon/issues#813] [Troubleshooting]
It seems that sometimes trains go through red presignal and cause deadlocks. I have relatively complicated rail system but I am relatively sure that my signaling is OK. After couple of hours there is lock situation and I do not see any other explanation than error in control logic. Is it known thing or should I try to find exact conditions and make some kind of bug report?

There is also another strange behavior with direction reversing. If train can find its destination without reversing it clearly prefers to go straight even it is very long and slow route. Usually I can control it by placing a waypoint near station in direction I want train to go. However, it does not work always. Sometimes the train continues forward from the station, use much time to find a loop and come back to go through the waypoint. For me the best solution would be ability to force reverse explicitly in schedule.

**Comment by DanielBarreleiro** [comment]
@hkoskenv Have you tried asking for help on discord?
I think it would be easier to help you out and share screenshots.

**Comment by xmnovotny** [comment]
1. Is it in the station with connected both ends bidirectionaly? If yes, this is not problem of pre signals - Assume that there is only one free track in the station and two trains comming in the oposite directions. For both trains pre-signals are yellow - track is free. When first train passes pre-signal, oposite pre-signal is still yellow, because station track is still free. The problem arises when the second train gets close enough to pass the pre-signal before the first train enters the station track.
2. You can force reversing in the schedule using settings button placed near the destination in the schedule task.
